server/aesdsocket.c

```diff
@@ -94,6 +219,16 @@ void process_client_connection(int connfd)
 cleanup:
 
     free(buff);
+    //acquire lock
+    if (pthread_mutex_lock(&queue_lock) != 0) {
+        syslog(LOG_ERR, "\r\n Failed to acquire lock"); //assert would be better
+    }
+    thread_params->task_finished = 1;
```

Comment:
@maheshchnt I don't think `queue_lock` is strictly necessary since:
1) You don't modify `connfd` outside the thread after pthread_create
2) You only set task_finished inside the thread and only check it outside.
Doesn't hurt anything to add it, just makes the code a bit more complex.